core/modules/ckeditor/ckeditor.theme.inc

```diff
@@ -62,9 +62,29 @@ function template_preprocess_ckeditor_settings_toolbar(&$variables) {
     elseif (isset($button['image_alternative'])) {
       $value = filter_xss_admin($button['image_alternative']);
     }
-    elseif (isset($button['image'])) {
-      $value = '<img src="'. file_create_url($button['image' . $rtl]) . '" title="' . check_plain($button['label']) . '" />';
-      $value = '<a href="#" role="button" title="' . $button['label'] . '" aria-label="' . $button['label'] . '"><span class="cke_button_icon">' . $value . '</span></a>';
+    elseif (isset($button['image']) || isset($button['image' . $rtl])) {
+      $button_image_path = isset($button['image' . $rtl]) ? $button['image' . $rtl] : $button['image'];
+      $button_label = check_plain($button['label']);
+
+      $button_image = theme('image', array('path' => $button_image_path, 'title' => $button_label,));
+      $button_image = '<span class="cke_button_icon">' . $button_image . '</span>';
+
+      // @see: theme_link().
+      $button_link = array();
+      $button_link['text'] = $button_image;
+      // 'text' contains HTML.
+      $button_link['options']['html'] = TRUE;
+      // Add "/#" as href.
+      $button_link['options']['fragment'] = FALSE;
+      // Remove leading "/" from href.
+      $button_link['options']['external'] = TRUE;
+      $button_link['options']['attributes'] = array(
+        'role' => 'button',
+        'aria-label' => $button_label,
+        'title' => $button_label,
+      );
+
+      $value = theme('link', $button_link);
```

Comment:
All of the setup for `theme('link')` here seems like it's a lot more finicky than just hand-coding the `<a>` tag.

Reply:
restored to hard-coded.